jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1964,25 +1966,66 @@ public URL getResource(String resName) {
 public InputStream getResourceAsStream(String resName) {
 	ClassLoader loader = this.getClassLoaderImpl();
 	String absoluteResName = this.toResourceName(resName);
+	InputStream result = null;
 /*[IF Sidecar19-SE]*/
 	Module thisModule = getModule();
 	
-	if (thisModule.isNamed() && (thisModule == System.getCallerClass().getModule())) {
+	if (useModularSearch(absoluteResName, thisModule,  System.getCallerClass())) {
 		try {
-			return thisModule.getResourceAsStream(absoluteResName);
+			result = thisModule.getResourceAsStream(absoluteResName);
 		} catch (IOException e) {
 			return null;
 		}
-	} else
+	}
+	if (null == result)
 /*[ENDIF] Sidecar19-SE */
 	{
 		if (loader == ClassLoader.bootstrapClassLoader) {
-			return ClassLoader.getSystemResourceAsStream(absoluteResName);
+			result =  ClassLoader.getSystemResourceAsStream(absoluteResName);
 		} else {
-			return loader.getResourceAsStream(absoluteResName);
+			result =  loader.getResourceAsStream(absoluteResName);
+		}
+	}
+	return result;
+}
+
+/*[IF Sidecar19-SE]*/
+/**
+ * Indicate if the package should be looked up in a module or via the class path.
+ * Look up the resource in the module if the module is named 
+ * and is the same module as the caller or the package is open to the caller.
+ * The default package (i.e. resources at the root of the module) is considered open.
+ * 
+ * @param absoluteResName name of resource, including package
+ * @param thisModule module of the current class
+ * @param callerClass class of method calling getResource() or getResourceAsStream()
+ * @return true if modular lookup should be used.
+ */
+private boolean useModularSearch(String absoluteResName, Module thisModule, Class callerClass) {
```

Comment:
`Class callerClass` -> `Class<?> callerClass`

Reply:
Right.  Done.
I will push the updates when my personal build completes.